gateway/src/shard/processor/impl.rs

```diff
@@ -78,23 +84,37 @@ impl ShardProcessor {
         });
 
         let session = Arc::new(Session::new(tx));
+        if resumable {
+            session
+                .id
+                .lock()
+                .await
+                .replace(config.session_id.clone().unwrap());
+            session
+                .seq
+                .store(config.sequence.unwrap(), Ordering::Relaxed)
+        }
 
         let (wtx, wrx) = watch_channel(Arc::clone(&session));
 
-        Ok((
-            Self {
-                config,
-                listeners: Listeners::default(),
-                properties,
-                rx,
-                session,
-                inflater: Inflater::new(shard_id),
-                url,
-                resume: None,
-                wtx,
-            },
-            wrx,
-        ))
+        let mut processor = Self {
+            config,
+            listeners: Listeners::default(),
+            properties,
+            rx,
+            session,
+            inflater: Inflater::new(shard_id),
+            url,
+            resume: None,
+            wtx,
+        };
+
+        if resumable {
+            debug!("Shard {:?} resuming", shard_id);
+            processor.resume().await;
```

Comment:
It's not clear what happens here when resuming fails, but my guess is that it'll probably already realise that from a session invalid response and make a new session? Or does it just stall? This should be documented

Reply:
the resume already handles it, if a session invalidates it gets requeued and identifies again, where should that exactly be documented tough? none of these functions have docs atm

Reply:
[Probably on top of the new function](https://github.com/twilight-rs/twilight/pull/188/files#diff-88211fc5e654c03fd0be06a8f305a9aeL58). It can be sort of like:
```rust
/// Creates a new shard processor.
///
/// If resume data is provided, but if the session has been invalidated due to too many events or too much time has passed, then a new session will automatically be created.
impl ShardProcessor { 
pub fn new(...) -> ...
```